Add informative Gravatar UI to the profile editing

hamorillo opened the pull request:
Fixes #20173

-----

We want to inform users that when editing a profile, they are actually editing their Gravatar profile. In this PR, we are adding an informative view for that.

We've created a similar [PR](https://github.com/wordpress-mobile/WordPress-iOS/pull/22563) on the iOS repo to keep both apps on a similar page.

The difference with the iOS app is that we'll show the Gravatar information banner in `MeFragment` instead of `MyProfileFragment`. That's because, in the Android app, users modify their avatar on the first screen, so we need to show the notice banner on that screen. 

Sync info banner will be shown in both fragments after any modification (avatar or field) is applied.

Jetpack:

| Dark Mode | Light Mode | Dark Mode Sync Message | Light Mode Sync Message | Dark - Profile Sync | Light - Profile Sync |
|----------|-----------|----------|-----------|----------|-----------|
|<img width="250" alt="image" src="https://github.com/wordpress-mobile/WordPress-Android/assets/6974554/0a3d0e87-9bbd-4e99-a60d-75c838477a9e">|<img width="250" alt="image" src="https://github.com/wordpress-mobile/WordPress-Android/assets/6974554/f0acc9b6-781f-49b1-abd5-907f2fdc183a">|<img width="250" alt="image" src="https://github.com/wordpress-mobile/WordPress-Android/assets/6974554/cc7fa030-ed7c-4204-81cc-292b5366392b">|<img width="250" alt="image" src="https://github.com/wordpress-mobile/WordPress-Android/assets/6974554/49ead560-aa3c-44ce-a74f-e415fdfbb3c8">|<img width="250" alt="image" src="https://github.com/wordpress-mobile/WordPress-Android/assets/6974554/f2e41964-bf7b-42e6-ba3a-9327e57e04e2">|<img width="250" alt="image" src="https://github.com/wordpress-mobile/WordPress-Android/assets/6974554/2165aa89-eba8-4cf0-8ece-2010282dfc83">|

WordPress:

| Dark Mode | Light Mode | Dark Mode Sync Message | Light Mode Sync Message | Dark - Profile Sync | Light - Profile Sync |
|----------|-----------|----------|-----------|----------|-----------|
|<img width="250" alt="image" src="https://github.com/wordpress-mobile/WordPress-Android/assets/6974554/2192601d-4ab7-4f53-b91f-3eaf1f78f792">|<img width="250" alt="image" src="https://github.com/wordpress-mobile/WordPress-Android/assets/6974554/c01a5685-6abc-4ec7-9518-dec0357eed82">|<img width="250" alt="image" src="https://github.com/wordpress-mobile/WordPress-Android/assets/6974554/625cc5b9-7e6f-4f4f-85c1-b951dcdbc694">|<img width="250" alt="image" src="https://github.com/wordpress-mobile/WordPress-Android/assets/6974554/2796ff7c-7d79-4bc1-863b-648c10886d4d">|<img width="250" alt="image" src="https://github.com/wordpress-mobile/WordPress-Android/assets/6974554/c96911a1-5cdc-4a31-8211-acba5b1157e3">|<img width="250" alt="image" src="https://github.com/wordpress-mobile/WordPress-Android/assets/6974554/9854e34d-a989-4cd0-8d14-ebad14e27ff4">|


## To Test:

Try on both JP and WP:

- Go to the `Me tab`
- Observe: There's a new informative UI as demonstrated above ☝️ (_If possible, verify in dark and light mode_)
- Click the link `Learn more on Gravatar.com` and verify you are redirected to `www.gravatar.com`
- Go back to the App
- Update your avatar
- Observe you see a notice message saying "Updates might take some time to sync with your Gravatar profile."
- Click the `Done` button and verify the notice message disappears
- Open `My Profile`
- Try updating those 4 fields
- Observe you see a notice message saying "Updates might take some time to sync with your Gravatar profile."
- Click the `Done` button and verify the notice message disappears

-----

## Regression Notes

1. Potential unintended areas of impact

    - We shouldn't impact other areas. We are adding views in `MyProfileFragment` and `MeFragment`, so all possible impacts should be inside that fragment.

2. What I did to test those areas of impact (or what existing automated tests I relied on)

    - N/A

3. What automated tests I added (or what prevented me from doing so)

    - N/A

-----

## PR Submission Checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

-----

## Testing Checklist:

- [ ] WordPress.com sites and self-hosted Jetpack sites.
- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] Talkback.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] Large and small screen sizes. (Tablet and smaller phones)
- [ ] Multi-tasking: Split screen and Pop-up view. (Android 10 or higher)

hamorillo committed:
Added Gravatar message in "My profile"

Showing the card with the Gravatar info in the My Profile fragment. The card should contain a link to the Gravartar.com website.

Design differs between light and dark modes.

hamorillo committed:
Added Gravatar Sync Info

When the user updates any of the fields, we need some time until the change is propagated to Gravatar. We want to inform the user about that.

Also, the user can dismiss that notification with a done button.

dangermattic commented:
<!--
  1 Error: PR requires at least one label...
  0 Warnings
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">PR requires at least one label.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

dangermattic commented:
<!--
  0 Errors
  1 Warning: PR is not assigned to a milest...
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">PR is not assigned to a milestone.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

Luchadores commented:
Hello, these look good too, I just wonder why is the Jetpack background fully white on White mode. For me, the screenshots I took look different where we have a slight gray background from the app itself. I will attach the image below for reference. 

![image](https://github.com/wordpress-mobile/WordPress-Android/assets/10038976/52dc0137-29c2-445b-b157-5619c187ca67)

hamorillo commented:
> Hello, these look good too, I just wonder why is the Jetpack background fully white on White mode. For me, the screenshots I took look different where we have a slight gray background from the app itself. I will attach the image below for reference.

@Luchadores If I'm not wrong, your screenshots were from the iOS app. The Android app has a fully white background, but I'm not sure why that difference, though. Anyway, I've used the already defined values in the app for the card background, so It should be good.

maxme commented:
I know it's not related to your code Hector, but the link to gravatar.com landing page is not a good experience:
- Mobile web margins seems broken (buttons are truncated).
- GDPR notice is truncated and the buttons are not visible.

https://github.com/wordpress-mobile/WordPress-Android/assets/40213/e4e9566a-f9e4-4c0f-9d0d-bb4633429001


---


I tried to find alternatives to the current link, but I couldn't find any better option. One alternative would be the "What's a Gravatar?" page from the FAQ but mobile web styling isn't really good either.

<img src=https://github.com/wordpress-mobile/WordPress-Android/assets/40213/5dce821c-9cfe-43d1-8ae9-cdd3052c6637 width=350/>


Conclusion to this: I think we should keep the link as it and fix the mobile web experience on gravatar.com. I'll open a ticket for that. cc @Luchadores

Luchadores commented:
Good point, @maxme. I agree, we should lead them to the same link, gravatar.com, and fix the mobile experience in a separate project/ticket.

ravishanker commented on WordPress/src/main/res/values/strings.xml:
```diff
@@ -2869,6 +2869,10 @@
     <string name="premium_upgrades_message">You have active premium upgrades on your site. Please cancel your upgrades prior to deleting your site.</string>
     <string name="show_purchases">Show purchases</string>
     <string name="checking_purchases">Checking purchases</string>
+    <string name="gravatar_info_description">"Gravatar keeps your profile information safe and up to date, automatically syncing any updates made here with your Gravatar profile."</string>
```
What are the quotes (") for in this string?

hamorillo replied:
Good catch. Removed! Thanks!

ravishanker reviewed (approved):
LGTM 🚀

ravishanker commented:
The english version of https://gravatar.com/ seems to be OK, probably other languages have issue on mobile web layout.
<img src="https://github.com/wordpress-mobile/WordPress-Android/assets/990349/c13d018c-88c6-4e05-8de4-899e187205cb" width=320 />

hamorillo committed:
Remove unnecessary quotes in string

maxme reviewed (changes requested):
Tested with the following contexts, everything looks good:
- [x] WordPress.com sites and self-hosted Jetpack sites.
- [x] Light and dark modes.
- [x] High contrast.
- [x] Talkback.
- [x] Languages with large words or with letters/accents not frequently used in English.
- [x] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [x] Large and small screen sizes. (Tablet and smaller phones)

There is an issue since the notice is not scrollable and can't be hidden, the profile fields are hidden and unusable. With the following contexts:
- [ ] Multi-tasking: Split screen and Pop-up view. (Android 10 or higher)
- [ ] Portrait and landscape orientations.
- [ ] Fonts: Larger, smaller and bold text.

## landscape
<img src=https://github.com/wordpress-mobile/WordPress-Android/assets/40213/0e775c35-680b-441a-8d00-02fd859225e2 width=300/>

## split screen
<img src=https://github.com/wordpress-mobile/WordPress-Android/assets/40213/8e9821b6-4d1c-48e7-ad4a-433b38b8c38b width=300/>

## bigger text size
<img src=https://github.com/wordpress-mobile/WordPress-Android/assets/40213/2c9beb9d-d588-452c-a410-9a590b62185a width=300/>


<img src=https://github.com/wordpress-mobile/WordPress-Android/assets/40213/923f30dc-8226-474c-a810-dabd31992359 width=300/>

hamorillo committed:
Fixing profile fields list constraint in MyProfileFragment

The list height was wrap_content, so when the space was limited with the new Gravatar view, the list was unusable. It kept behind the Gravatar view.

This was visible, for example, in landscape or with a bigger text size.

Adding the constraint to the top of the Gravatar view should minimize the issue.

There are still some corners case where the UX can be improved.

hamorillo commented:
> There is an issue since the notice is not scrollable and can't be hidden, the profile fields are hidden and unusable. 

❤️ Great catch @maxme! I've fixed the constraint of `nested_scroll_view` but there are some corner cases where the UX can still be improved. However, for those cases, I think we need to talk with designers about the possibility of making the notice hideable. Not sure if those are blocking, though. Wdyt?

<img width="2402" alt="image" src="https://github.com/wordpress-mobile/WordPress-Android/assets/6974554/e758f99a-9bc4-4bfb-929f-61c07d0422a7">

maxme reviewed (approved):
> I've fixed the constraint of nested_scroll_view but there are some corner cases where the UX can still be improved.

Works fine, the view is not ideal when hidden behind the notice but still usable, so :shipit: 

> Not sure if those are blocking, though. Wdyt?

Not blocking IMO there are parts of the app that won't work when you're in landscape + huge text.

hamorillo committed:
Move Gravatar info banner from "MyProfileFragment" to "MeFragment"

After reviewing with designers, we want to show the informative banner on the first screen as users can modify their avatar in the "Profile" fragment.

The sync information banner will be shown in both fragments, MyProfile and MeFragment, after any modification on the Gravatar profile. The sync banner is reused, so we've extracted it to a separate XML layout.

This commit mainly moves the logic and view from the MeFragment to the MyProfile one.

dangermattic commented:
<!--
  0 Errors
  3 Warnings: <code>strings.xml</code> files...
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          3 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><code>strings.xml</code> files should only be updated on release branches, when the translations are downloaded by our automation.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 300 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/wordpress-mobile/WordPress-Android/milestone/277">24.3</a>. The due date for this milestone has already passed.<br />
Please assign it to a milestone with a later deadline or check whether the release for this milestone has already been finished.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

hamorillo commented:
@ravishanker @maxme, Sorry for bothering you again. I've moved the Gravatar info banner from `MyProfileFragment` to `MeFragment`. I've also updated the PR description. Can you do another round?

maxme commented:
Code and UI looks good to me, but now I have a doubt about the copy, specifically the mention to "WordPress.com". When I run the Jetpack variant: 

![Screenshot_20240216_144359](https://github.com/wordpress-mobile/WordPress-Android/assets/40213/28d9729c-b046-49d8-86fc-efdfc0a39c8d)

@ravishanker maybe you can help, is this correct to mention "WordPress.com profile"on the Jetpack variant? Should we change the copy? I think it's fine since the login screens mention a "WordPress.com account" as well:

![Screenshot_20240216_145504](https://github.com/wordpress-mobile/WordPress-Android/assets/40213/79de1d85-8631-4cbd-afa9-8b1edd21cf45)

Luchadores commented:
Uuu, good catch. Could you tag Ronnie, I cannot seem to find his handle here. @maxme

ravishanker commented:
> is this correct to mention "WordPress.com profile"on the Jetpack variant

Yes, that's correct.  It is a [WordPress.com](https://wordpress.com/) account

If anything, the `About Jetpack`, `Log out of Jetpack` should probably just read `About`, and `Logout`.  Not for this PR though.

ravishanker commented:
Me screen is a better location than profile screen.  Debug Settings, and Design System options there won't appear on Release version, so it should appear in full above the bottom nav.  However
- The background contrast is now much lighter now than before in both light, and dark theme.  Is that intentional?  
- The title (Your WordPress.com....) font is too bold, and shouting, feels out of place with the rest of the app.  Look at My Site and else where for reference

maxme commented:
> Me screen is a better location than profile screen. Debug Settings, and Design System options there won't appear on Release version, so it should appear in full above the bottom nav. However
> 
>   - The background contrast is now much lighter now than before in both light, and dark theme.  Is that intentional?
> 
>   - The title (Your WordPress.com....) font is too bold, and shouting, feels out of place with the rest of the app.  Look at My Site and else where for reference

@Luchadores what do you recommend with the title font weight and the background color?

maxme commented:
Updated with a smaller text size for the title

Light|Dark
-|-
![Screenshot_20240219_093114](https://github.com/wordpress-mobile/WordPress-Android/assets/40213/aab30572-7d3e-4e90-851b-51349d85e620)|![Screenshot_20240219_093054](https://github.com/wordpress-mobile/WordPress-Android/assets/40213/04bcbfc8-ece6-4f54-8bdd-796996a18e5b)

maxme commented:
Darker background `#F2F2F7` for light theme:

<img width=300 src=https://github.com/wordpress-mobile/WordPress-Android/assets/40213/70e0a765-bf4f-44ab-b3b4-f6490154acf6/>

Dark theme was already using `#1C1C1E`, not sure if you want to change this one.

mlumeau commented:
> Updated with a smaller text size for the title

I think it's a bit better, minus the orphan ("Gravatar" alone in a new line).
> Darker background #F2F2F7 for light theme

I think it's much better against the Android app's white background
cc: @Luchadores

Luchadores commented:
👋 

Thank you all for looking at this, it's much appreciated. I think colors are fine, as we have to have a bit of contrast between the page background and the notice after updating your profile. 

We can also go with the text size from Maxime if that fits better. I don't have an Android phone, so I always have to ask for screenshots of existing UI.

PS: I would force a line break so that we don't end up having orphans like we have in the screenshot with "Gravatar"
![image](https://github.com/wordpress-mobile/WordPress-Android/assets/10038976/f72c4838-516d-4228-aa50-1dd05524054a)

Thank you!
@mlumeau @maxme

maxme commented:
We don't need to force a line break (this would look weird in landscape mode or on a tablet). I fixed it throught WPTextView in 1e331558d7940e6d431aef4ff14aa1ee0e943c7a, it replaces the last space of a sentence by a non breakable space.

<img width=300 src=https://github.com/wordpress-mobile/WordPress-Android/assets/40213/7d9eb23b-b223-4e4c-95ad-fca0ff9f36fb/>

maxme committed:
Fix for widow words

maxme committed:
Update Gravatar info text size and background color on light theme

mlumeau commented:
> I fixed it throught WPTextView in https://github.com/wordpress-mobile/WordPress-Android/commit/1e331558d7940e6d431aef4ff14aa1ee0e943c7a, it replaces the last space of a sentence by a non breakable space.

TIL!
